components/ai_chat/core/browser/constants.h

```diff
@@ -24,12 +25,21 @@ extern const base::fixed_flat_set<std::string_view, 1>
 inline constexpr uint8_t kMaxPreviewPages = 20;
 extern const char kLeoModelSupportUrl[];
 
-// Llama 3 has a max context length of 8k tokens. Phi 3 Mini and Llama 2 have 4k
-// tokens max. Using Phi 3 Mini as the upper bound, 1 token ≈ 4 characters, so
-// the max context length is ~16k characters. To allow for follow-ups, set max
-// page content length to 60% of this, i.e., 9.6k characters.
-inline constexpr int kCustomModelMaxPageContentLength = 9600;
-inline constexpr int kCustomModelLongConversationCharLimit = 10000;
+/**
+ * Upon registering a custom model, users have the ability to explicitly
+ * provide a context size (in tokens). When present, we'll use this value
+ * to determine the max page content length (in chars). We will assume 4
+ * chars per token. When no context size has been provided, we will
+ * default to a conservative 4k tokens based on common models like Phi 3
+ * Mini and Llama 2 (both have 4k token context limits).
+ */
+inline constexpr size_t kDefaultCharsPerToken = 4;
+inline constexpr float kMaxContentLengthThreshold = 0.6f;
+inline constexpr size_t kCustomModelDefaultContextSize = 4000;
```

Comment:
All of these are trivial values and don't need to be `inline constexpr T`. They can be just `constexpr T` as it doesn't matter if the compiler makes copies of them.

Reply:
I'm inclined to leave them as `inline` since that predates my PR, and it seems to add value (albeit very little) since it enables optimizations for the compiler.

Reply:
It doesn't enable any compiler optimisations for trivial types.

Reply:
@jonathansampson please make sure you have an actual resolution with the person who made the comments before resolving them. Either make appropriate changes or the person who made the comment can resolve it without changes. In this case I agree with @cdesouza-chromium that since you're touching these anyway we should remove `inline`

Reply:
This request of mine was wrong. In practice there's no difference performance-wise if this value is `constepxr`, or `inline constexpr`. However, `constexpr` values do have internal linkage, and these are declarations on a header. Without the `inline` specifier, these constants will have a new address for each translation unit. This is not a problem most of the time, as these are trivial values, that we just copy them , but [it will be UB](https://eel.is/c++draft/dcl.constexpr#1.sentence-3) if someone takes these values by reference, as it violates the One Definition Rule. I found out about this only this week. Since then I found out a [Tip Of The Week](https://abseil.io/tips/140) that instructs to always use `inline` for `constexpr` constants in header files.

I'm working on a PR to fix all places on our codebase.

I apologise for the bad advice I provided in this instance, and for insisting on it.

cc @jonathansampson @bridiver.

[1] https://eel.is/c++draft/dcl.constexpr#1.sentence-3
[2] https://abseil.io/tips/140